JustBrewIt/Pages/Search.cshtml.cs

```diff
@@ -1,59 +1,84 @@
+        public bool IsSearchType { get; set; }
         public string Url { get; set; }
-
+        public bool IsSearchValid = true;
+        public bool IsSearchEmpty = false;
         public void OnGet()
         {
-            isSearchCity = false;
-            isSearchType = false;
+            IsSearchCity = false;
+            IsSearchType = false;
         }
         
         public void OnPost()
         {
-            string city = BreweryCity;
-            string type = BreweryType;
+            try
+            {
+                if (string.IsNullOrEmpty(BreweryCity) && string.IsNullOrEmpty(BreweryType))
+                {
+                    IsSearchEmpty = true;
+                }
+                else
+                {
+                    string city = BreweryCity;
+                    string type = BreweryType;
 
-           if(city == null && type != null) { 
+                    var baseUri = _configuration["BreweryBaseUri"];
 
-            Url = "https://api.openbrewerydb.org/breweries?by_state=ohio&by_type=" + BreweryType;
-           }
-           else if(city != null && type != null)
-            {
-                Url = "https://api.openbrewerydb.org/breweries?by_type=" + type + "&by_city=" + city;
-            }
-            else
-            {
-                Url = "https://api.openbrewerydb.org/breweries?by_city=" + city;
+                    if (city == null && type != null)
+                    {
+                        Url = baseUri + "?by_state=ohio&by_type=" + BreweryType;
+                    }
+                    else if (city != null && type != null)
+                    {
+                        Url = baseUri + "?by_type=" + type + "&by_city=" + city;
+                    }
+                    else
+                    {
+                        Url = baseUri + "?by_city=" + city;
+                    }
+
+                    using (var webClient = new WebClient())
+                    {
+                        string jsonString = webClient.DownloadString(Url);
+                        Welcome[] welcome = Welcome.FromJson(jsonString);
+                        if (welcome.Length != 0)
+                            ViewData["BreweryAPI"] = welcome;
+                        else
+                            IsSearchValid = false;
+
+                    }
+
+                    IsSearchType = true;
+                    IsSearchCity = true;
+                }
             }
-            
-            using (var webClient = new WebClient())
+            catch(Exception ex)
             {
-                string jsonString = webClient.DownloadString(Url);
-                Welcome[] welcome = Welcome.FromJson(jsonString);
-                ViewData["BreweryAPI"] = welcome;
+                //this should be handled through application's exception handling mechanism or redirect to error page
+                Console.WriteLine("Exception occured while searching the brewery");
```

Comment:
Exception handling: Good idea.

Logging: Good idea as well.  It is more precise than Console.WriteLine()

Putting conversation/discussion  in code comments is bad practice, as that clutters up the source code and does not belong in source code files.  I mentioned that in the code review instructions.    Discussion and comments are very good, but have that in the appropriate place, which is in the "Files Changed" tab in the pull request in GitHub (which is where I am writing this and you are reading this).  That way, you can have a conversation with replies, right where the code is happening, and without cluttering up the source code itself.